candid/tests/parse_type.rs

```diff
@@ -12,7 +12,7 @@ fn parse_idl_prog() {
     let prog = r#"
 import "test.did";
 type my_type = nat8;
-type List = record { head: int; tail: List };
+type List = record { head: int; tail: opt List };
```

Comment:
```suggestion
type List = variant { head: int; tail: List };
```

Reply:
This change doesn't make sense to me. That variant doesn't define a list, whereas the record did define a (non-empty) list.

Reply:
Right, I wasn't thinking. As @nomeata suggested, it should be `opt record` instead.